### PR TITLE
Editing pass on starter template readme

### PR DESCRIPTION
**Pull request by pdg744**
- updates link to new version of "Understanding the Starter Template"
- updates bonsai access pointer to `bonsai.xyz/apply` (this link isn't live yet but should be later this week)
- addresses the fact that the TODOs at the top of the README render as a single line in the markdown display version
- updates link from "Getting Started" section on website to zkvm dev docs

### CODE REVIEW

**Comment by pdg744** [comment]
This shouldn't land until https://bonsai.xyz/apply is live

File: templates/rust-starter/README.md [hunk 1]
```diff
@@ -50,4 +51,4 @@ For a walk-through of how to build with this template, check out this [excerpt f
 ## Questions, Feedback, and Collaborations
 We'd love to hear from you on [Discord](https://discord.gg/risczero) or [Twitter](https://twitter.com/risczero).
 
-[Understanding the Starter Template]: https://www.risczero.com/docs/examples/understanding_template
+[Structure of a zkVM Application]: https://www.dev.risczero.com/zkvm/developer-guide/zkvm-app-structure
```
Comment (tzerrell) [inline templates/rust-starter/README.md:54]: The `www.` is unnecessary and I think it's nicer without it.
```suggestion
[Structure of a zkVM Application]: https://dev.risczero.com/zkvm/developer-guide/zkvm-app-structure
```

File: templates/rust-starter/README.md [hunk 2]
```diff
@@ -1,10 +1,11 @@
 # RISC Zero Rust Starter Template
 
 Welcome to the RISC Zero Rust Starter Template! This template is intended to give you a starting point for building a project using the RISC Zero zkVM. Throughout the code are comments labelled `TODO` in places where we expect projects will need to modify the code.
```
Comment (tzerrell) [inline templates/rust-starter/README.md:3]: Reviewing this makes me realize the original language is potentially confusion, is this clearer?
```suggestion
Welcome to the RISC Zero Rust Starter Template! This template is intended to give you a starting point for building a project using the RISC Zero zkVM. Throughout the code are comments labelled `TODO` in places where we expect projects will need to modify the code (including this document!).
```

**Review by tzerrell — COMMENTED** [review]
Looks good

File: templates/rust-starter/README.md (hunk 2 above)
Comment (pdg744) [reply templates/rust-starter/README.md:3]: How about this? 
```suggestion
Welcome to the RISC Zero Rust Starter Template! This template is intended to give you a starting point for building a project using the RISC Zero zkVM. Throughout the template (including in this README), you'll find comments labelled `TODO` in places where you'll need to make changes.
```

**Review by tzerrell — APPROVED** [review]

**Pull request merged**